nextcord/interactions.py

```diff
@@ -1252,3 +1257,51 @@ class InteractionMessage(_InteractionMessageMixin, Message):
     """
 
     pass
+
+
+class MessageInteraction:
+    """Represents a message's interaction data, regardless of application.
+
+    A message's interaction data is a property of a message when the message
+    is a response to an interaction from any bot.
+
+    .. versionadded:: 2.0
+
+    Attributes
+    -----------
+    data: :class:`dict`
+        The raw data from the interaction.
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    name: :class:`str`
+        The application ID that the interaction was for.
+    user: :class:`User`
+        The user that sent the interaction.
+    member: Optional[:class:`Member`]
+        The member that sent the interaction,
+        if the interaction occured in a guild.
```

Comment:
For consistency with Interaction, this should probably be just under `MessageInteraction.user` and just make it sometimes a `User` and sometimes a `Member`